Core/Projectiles/Projectile_Manager.py

```diff
@@ -0,0 +1,95 @@
+import arcade
+import math
+
+from Constants.Physics import PLAYER_MOVEMENT_SPEED, BULLET_MOVE_FORCE
+
+
+class ProjectileManager:
+    def __init__(self, game_resources):
+
+        self.game_resources = game_resources
+
+        self.projectile_physics = arcade.PymunkPhysicsEngine()
+        self.projectile_physics.add_sprite_list(
+            self.game_resources.wall_list,
+            collision_type="wall",
+            body_type=arcade.PymunkPhysicsEngine.STATIC,
+        )
+
+        def wall_hit_handler(bullet_sprite, _wall_sprite, _arbiter, _space, _data):
+            """ Called for bullet/wall collision """
+            bullet_sprite.remove_from_sprite_lists()
+
+        self.projectile_physics.add_collision_handler(
+            "bullet", "wall", post_handler=wall_hit_handler
+        )
+
+        '''
+        def item_hit_handler(bullet_sprite, item_sprite, _arbiter, _space, _data):
+            """ Called for bullet/wall collision """
+            bullet_sprite.remove_from_sprite_lists()
+            item_sprite.remove_from_sprite_lists()
+
+        self.projectile_physics.add_collision_handler("bullet", "item", post_handler=item_hit_handler)
+        '''
+
+    def on_mouse_press(self, x, y, button, modifiers):
+        """ Called whenever the mouse button is clicked. """
+
+        bullet = BulletSprite(20, 5, arcade.color.DARK_YELLOW)
+        self.game_resources.bullet_list.append(bullet)
+
+        # Position the bullet at the player's current location
+        start_x = self.game_resources.player_sprite.center_x
+        start_y = self.game_resources.player_sprite.center_y
+        bullet.position = self.game_resources.player_sprite.position
+
+        # Get from the mouse the destination location for the bullet
+        # IMPORTANT! If you have a scrolling screen, you will also need
+        # to add in self.view_bottom and self.view_left.
+        dest_x = x
+        dest_y = y
+
+        # Do math to calculate how to get the bullet to the destination.
+        # Calculation the angle in radians between the start points
+        # and end points. This is the angle the bullet will travel.
+        x_diff = dest_x - start_x
+        y_diff = dest_y - start_y
+        angle = math.atan2(y_diff, x_diff)
```

Comment:
Nice maths here. Did you figure this out yourself or were you able to use a guide? Just curious!